### PR TITLE
Add an about dialog with the version number

### LINKED ISSUE #867: Feature Request: make the version number visible somewhere in the UI (menu maybe?)

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request
A picture's worth a thousand words:
image

## PR Checklist
* [x] Closes #867
* [x] I've discussed this with @DHowett-MSFT

### CODE REVIEW

File: src/cascadia/TerminalApp/App.cpp [hunk 1]
```diff
@@ -362,6 +406,18 @@ namespace winrt::TerminalApp::implementation
 
             feedbackFlyout.Click({ this, &App::_FeedbackButtonOnClick });
             newTabFlyout.Items().Append(feedbackFlyout);
+
+            // Create the about button.
+            auto aboutFlyout = Controls::MenuFlyoutItem{};
+            aboutFlyout.Text(L"About");
+
+            Controls::FontIcon aboutIco{};
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/App.cpp]: Possibly prefer `Symbol::Help()`

File: src/cascadia/TerminalApp/App.cpp [hunk 2]
```diff
@@ -218,6 +213,55 @@ namespace winrt::TerminalApp::implementation
...
+        const auto package = winrt::Windows::ApplicationModel::Package::Current();
+        const auto packageName = package.DisplayName();
+        const auto version = package.Id().Version();
+        std::wstringstream aboutTextStream;
+
+        // Format our about text. It will look like the following:
+        // <Display Name>
+        // Version: <Major>.<Minor>.<Build>.<Revision>
+
+        aboutTextStream << packageName.c_str() << L"\n";
```
Comment (adiviness) [inline src/cascadia/TerminalApp/App.cpp]: must these call `c_str()`?

File: src/cascadia/TerminalApp/App.cpp [hunk 3]
```diff
@@ -362,6 +406,18 @@ namespace winrt::TerminalApp::implementation
 
             feedbackFlyout.Click({ this, &App::_FeedbackButtonOnClick });
             newTabFlyout.Items().Append(feedbackFlyout);
+
+            // Create the about button.
+            auto aboutFlyout = Controls::MenuFlyoutItem{};
+            aboutFlyout.Text(L"About");
+
+            Controls::FontIcon aboutIco{};
+            aboutIco.Glyph(L"\xF142");
```
Comment (adiviness) [inline src/cascadia/TerminalApp/App.cpp]: can this have a name? perhaps in unicode.hpp?

File: src/cascadia/TerminalApp/App.cpp [hunk 4]
```diff
@@ -405,6 +461,14 @@ namespace winrt::TerminalApp::implementation
         winrt::Windows::System::Launcher::LaunchUriAsync({ L"https://github.com/microsoft/Terminal/issues" });
     }
 
+    // Method Description:
```
Comment (adiviness) [inline src/cascadia/TerminalApp/App.cpp]: nit: missing args doc comments

File: src/cascadia/TerminalApp/App.cpp (hunk 3 above)
Comment (DHowett-MSFT) [reply src/cascadia/TerminalApp/App.cpp]: This will be helped by using `Symbol::Help()`

File: src/cascadia/TerminalApp/App.cpp (hunk 2 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/App.cpp]: Unfortunately yes, since these are hstrings, which don't work with wstringstream's `operator<<`

**Comment by mdtauk** [comment]
When the Settings UI is added, there could be an about section there, with a change log and version information.

Rather than spawning different dialogs for About, What's new, and Settings - keep it contained in a settings dialog, which has sections for the various settings...

image

**Comment by DHowett-MSFT** [comment]
Yes, we're not going to just keep adding things to the menu once we have some ancillary UI support. Thanks!

**Pull request merged**